Set SameSite=None cookies from a secure origin in http/tests/resourceLoadStatistics/

charliewolfe opened the pull request:
#### 6b99c648fb7c300ff5ec61f9b80ca78812ad8174
<pre>
Set SameSite=None cookies from a secure origin in http/tests/resourceLoadStatistics/
<a href="https://bugs.webkit.org/show_bug.cgi?id=275747">https://bugs.webkit.org/show_bug.cgi?id=275747</a>
<a href="https://rdar.apple.com/129872343">rdar://129872343</a>

Reviewed by Matthew Finkel and Sihui Liu.

* LayoutTests/http/tests/resourceLoadStatistics/add-blocking-to-redirect.html:
* LayoutTests/http/tests/resourceLoadStatistics/blocking-in-web-worker-script-import.https-expected.txt: Renamed from LayoutTests/http/tests/resourceLoadStatistics/blocking-in-web-worker-script-import-expected.txt.
* LayoutTests/http/tests/resourceLoadStatistics/blocking-in-web-worker-script-import.https.html: Renamed from LayoutTests/http/tests/resourceLoadStatistics/blocking-in-web-worker-script-import.html.
* LayoutTests/http/tests/resourceLoadStatistics/cname-cloaking-top-cname-sub-1p-cname.https-expected.txt: Renamed from LayoutTests/http/tests/resourceLoadStatistics/cname-cloaking-top-cname-sub-1p-cname-expected.txt.
* LayoutTests/http/tests/resourceLoadStatistics/cname-cloaking-top-cname-sub-1p-cname.https.html: Renamed from LayoutTests/http/tests/resourceLoadStatistics/cname-cloaking-top-cname-sub-1p-cname.html.
* LayoutTests/http/tests/resourceLoadStatistics/cname-cloaking-top-cname-sub-3p-cname.https-expected.txt: Renamed from LayoutTests/http/tests/resourceLoadStatistics/cname-cloaking-top-cname-sub-3p-cname-expected.txt.
* LayoutTests/http/tests/resourceLoadStatistics/cname-cloaking-top-cname-sub-3p-cname.https.html: Renamed from LayoutTests/http/tests/resourceLoadStatistics/cname-cloaking-top-cname-sub-3p-cname.html.
* LayoutTests/http/tests/resourceLoadStatistics/cname-cloaking-top-cname-sub-matching-cname.https-expected.txt: Renamed from LayoutTests/http/tests/resourceLoadStatistics/cname-cloaking-top-cname-sub-matching-cname-expected.txt.
* LayoutTests/http/tests/resourceLoadStatistics/cname-cloaking-top-cname-sub-matching-cname.https.html: Renamed from LayoutTests/http/tests/resourceLoadStatistics/cname-cloaking-top-cname-sub-matching-cname.html.
* LayoutTests/http/tests/resourceLoadStatistics/cname-cloaking-top-cname-sub-no-cname.https-expected.txt: Renamed from LayoutTests/http/tests/resourceLoadStatistics/cname-cloaking-top-cname-sub-no-cname-expected.txt.
* LayoutTests/http/tests/resourceLoadStatistics/cname-cloaking-top-cname-sub-no-cname.https.html: Renamed from LayoutTests/http/tests/resourceLoadStatistics/cname-cloaking-top-cname-sub-no-cname.html.
* LayoutTests/http/tests/resourceLoadStatistics/cname-cloaking-top-no-cname-sub-1p-cname.https-expected.txt: Renamed from LayoutTests/http/tests/resourceLoadStatistics/cname-cloaking-top-no-cname-sub-1p-cname-expected.txt.
* LayoutTests/http/tests/resourceLoadStatistics/cname-cloaking-top-no-cname-sub-1p-cname.https.html: Renamed from LayoutTests/http/tests/resourceLoadStatistics/cname-cloaking-top-no-cname-sub-1p-cname.html.
* LayoutTests/http/tests/resourceLoadStatistics/cname-cloaking-top-no-cname-sub-3p-cname.https-expected.txt: Renamed from LayoutTests/http/tests/resourceLoadStatistics/cname-cloaking-top-no-cname-sub-3p-cname-expected.txt.
* LayoutTests/http/tests/resourceLoadStatistics/cname-cloaking-top-no-cname-sub-3p-cname.https.html: Renamed from LayoutTests/http/tests/resourceLoadStatistics/cname-cloaking-top-no-cname-sub-3p-cname.html.
* LayoutTests/http/tests/resourceLoadStatistics/cname-cloaking-top-no-cname-sub-no-cname.https-expected.txt: Renamed from LayoutTests/http/tests/resourceLoadStatistics/cname-cloaking-top-no-cname-sub-no-cname-expected.txt.
* LayoutTests/http/tests/resourceLoadStatistics/cname-cloaking-top-no-cname-sub-no-cname.https.html: Renamed from LayoutTests/http/tests/resourceLoadStatistics/cname-cloaking-top-no-cname-sub-no-cname.html.
* LayoutTests/http/tests/resourceLoadStatistics/cookie-deletion.html:
* LayoutTests/http/tests/resourceLoadStatistics/cookies-with-and-without-user-interaction.html:
* LayoutTests/http/tests/resourceLoadStatistics/exemptDomains/app-bound-domains-exempt-from-cookie-blocking-between-each-other.html:
* LayoutTests/http/tests/resourceLoadStatistics/exemptDomains/managed-domains-third-party-resources-exempt-from-cookie-blocking.html:
* LayoutTests/http/tests/resourceLoadStatistics/exemptDomains/third-party-cookie-blocking.html:
* LayoutTests/http/tests/resourceLoadStatistics/grandfathering.html:
* LayoutTests/http/tests/resourceLoadStatistics/no-third-party-cookie-blocking-when-itp-is-off.html:
* LayoutTests/http/tests/resourceLoadStatistics/non-prevalent-resources-can-access-cookies-in-a-third-party-context.html:
* LayoutTests/http/tests/resourceLoadStatistics/ping-to-prevalent-resource.html:
* LayoutTests/http/tests/resourceLoadStatistics/resources/secure-worker-importing-localhost-script.js: Added.
(onmessage):
* LayoutTests/http/tests/resourceLoadStatistics/resources/set-cookie.py:
* LayoutTests/http/tests/resourceLoadStatistics/third-party-cookie-blocking-on-sites-without-user-interaction.html:
* LayoutTests/platform/glib/TestExpectations:
* LayoutTests/platform/gtk/TestExpectations:
* LayoutTests/platform/ios/TestExpectations:
* LayoutTests/platform/mac-wk2/TestExpectations:
* LayoutTests/platform/wincairo/TestExpectations:
* LayoutTests/platform/wk2/TestExpectations:

Canonical link: <a href="https://commits.webkit.org/280300@main">https://commits.webkit.org/280300@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/1c207a27ea79d49bb742033dcdf7ee384b9051c3

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/56211 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/35537 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/8683 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/59817 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/6647 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/43159 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/6841 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/45471 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/4619 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/58240 "Passed tests") | [❌ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/33424 "Found 60 new test failures: accessibility/accessibility-node-reparent.html accessibility/ancestor-computation.html accessibility/announcement-notification.html accessibility/aria-braillelabel.html accessibility/aria-brailleroledescription.html accessibility/aria-checked-mixed-value.html accessibility/aria-current-state-changed-notification.html accessibility/aria-current.html accessibility/aria-describedby-on-input.html accessibility/aria-description.html ... (failure)") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/48494 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/26384 "Passed tests") | 
| | [❌ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/30204 "Found 60 new test failures: imported/w3c/web-platform-tests/FileAPI/Blob-methods-from-detached-frame.html imported/w3c/web-platform-tests/FileAPI/FileReader/Progress_event_bubbles_cancelable.html imported/w3c/web-platform-tests/FileAPI/FileReader/workers.html imported/w3c/web-platform-tests/FileAPI/FileReaderSync.worker.html imported/w3c/web-platform-tests/FileAPI/blob/Blob-array-buffer.any.html imported/w3c/web-platform-tests/FileAPI/blob/Blob-array-buffer.any.worker.html imported/w3c/web-platform-tests/FileAPI/blob/Blob-bytes.any.html imported/w3c/web-platform-tests/FileAPI/blob/Blob-bytes.any.worker.html imported/w3c/web-platform-tests/FileAPI/blob/Blob-constructor-dom.window.html imported/w3c/web-platform-tests/FileAPI/blob/Blob-constructor-endings.html ... (failure)") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/5824 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/5651 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/52211 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/6096 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/61500 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/119 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/6220 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/52777 "Passed tests") | 
| [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🧪 services](https://ews-build.webkit.org/#/builders/28/builds/55903 "Build is in progress. Recent messages:") | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/119 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/48561 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/52630 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/105 "Passed tests") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/8340 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/31364 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/32450 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/33533 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/32197 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

sysrqb commented on LayoutTests/http/tests/resourceLoadStatistics/add-blocking-to-redirect.html:
```diff
@@ -11,7 +11,7 @@
 
     const partitionHost = "127.0.0.1:8000";
-    const thirdPartyOrigin = "http://localhost:8000";
+    const thirdPartyOrigin = "https://localhost:8443";
```
Should we explicitly set samesite=none on the cookies (e.g., `firstPartyCookieName + "&value=value;samesite=none"` below?

charliewolfe replied:
SameSite=None is explicitly set in set-cookie.py

charliewolfe committed:
Set SameSite=None cookies from a secure origin in http/tests/resourceLoadStatistics/
https://bugs.webkit.org/show_bug.cgi?id=275747
rdar://129872343

Reviewed by Matthew Finkel and Sihui Liu.

* LayoutTests/http/tests/resourceLoadStatistics/add-blocking-to-redirect.html:
* LayoutTests/http/tests/resourceLoadStatistics/blocking-in-web-worker-script-import.https-expected.txt: Renamed from LayoutTests/http/tests/resourceLoadStatistics/blocking-in-web-worker-script-import-expected.txt.
* LayoutTests/http/tests/resourceLoadStatistics/blocking-in-web-worker-script-import.https.html: Renamed from LayoutTests/http/tests/resourceLoadStatistics/blocking-in-web-worker-script-import.html.
* LayoutTests/http/tests/resourceLoadStatistics/cname-cloaking-top-cname-sub-1p-cname.https-expected.txt: Renamed from LayoutTests/http/tests/resourceLoadStatistics/cname-cloaking-top-cname-sub-1p-cname-expected.txt.
* LayoutTests/http/tests/resourceLoadStatistics/cname-cloaking-top-cname-sub-1p-cname.https.html: Renamed from LayoutTests/http/tests/resourceLoadStatistics/cname-cloaking-top-cname-sub-1p-cname.html.
* LayoutTests/http/tests/resourceLoadStatistics/cname-cloaking-top-cname-sub-3p-cname.https-expected.txt: Renamed from LayoutTests/http/tests/resourceLoadStatistics/cname-cloaking-top-cname-sub-3p-cname-expected.txt.
* LayoutTests/http/tests/resourceLoadStatistics/cname-cloaking-top-cname-sub-3p-cname.https.html: Renamed from LayoutTests/http/tests/resourceLoadStatistics/cname-cloaking-top-cname-sub-3p-cname.html.
* LayoutTests/http/tests/resourceLoadStatistics/cname-cloaking-top-cname-sub-matching-cname.https-expected.txt: Renamed from LayoutTests/http/tests/resourceLoadStatistics/cname-cloaking-top-cname-sub-matching-cname-expected.txt.
* LayoutTests/http/tests/resourceLoadStatistics/cname-cloaking-top-cname-sub-matching-cname.https.html: Renamed from LayoutTests/http/tests/resourceLoadStatistics/cname-cloaking-top-cname-sub-matching-cname.html.
* LayoutTests/http/tests/resourceLoadStatistics/cname-cloaking-top-cname-sub-no-cname.https-expected.txt: Renamed from LayoutTests/http/tests/resourceLoadStatistics/cname-cloaking-top-cname-sub-no-cname-expected.txt.
* LayoutTests/http/tests/resourceLoadStatistics/cname-cloaking-top-cname-sub-no-cname.https.html: Renamed from LayoutTests/http/tests/resourceLoadStatistics/cname-cloaking-top-cname-sub-no-cname.html.
* LayoutTests/http/tests/resourceLoadStatistics/cname-cloaking-top-no-cname-sub-1p-cname.https-expected.txt: Renamed from LayoutTests/http/tests/resourceLoadStatistics/cname-cloaking-top-no-cname-sub-1p-cname-expected.txt.
* LayoutTests/http/tests/resourceLoadStatistics/cname-cloaking-top-no-cname-sub-1p-cname.https.html: Renamed from LayoutTests/http/tests/resourceLoadStatistics/cname-cloaking-top-no-cname-sub-1p-cname.html.
* LayoutTests/http/tests/resourceLoadStatistics/cname-cloaking-top-no-cname-sub-3p-cname.https-expected.txt: Renamed from LayoutTests/http/tests/resourceLoadStatistics/cname-cloaking-top-no-cname-sub-3p-cname-expected.txt.
* LayoutTests/http/tests/resourceLoadStatistics/cname-cloaking-top-no-cname-sub-3p-cname.https.html: Renamed from LayoutTests/http/tests/resourceLoadStatistics/cname-cloaking-top-no-cname-sub-3p-cname.html.
* LayoutTests/http/tests/resourceLoadStatistics/cname-cloaking-top-no-cname-sub-no-cname.https-expected.txt: Renamed from LayoutTests/http/tests/resourceLoadStatistics/cname-cloaking-top-no-cname-sub-no-cname-expected.txt.
* LayoutTests/http/tests/resourceLoadStatistics/cname-cloaking-top-no-cname-sub-no-cname.https.html: Renamed from LayoutTests/http/tests/resourceLoadStatistics/cname-cloaking-top-no-cname-sub-no-cname.html.
* LayoutTests/http/tests/resourceLoadStatistics/cookie-deletion.html:
* LayoutTests/http/tests/resourceLoadStatistics/cookies-with-and-without-user-interaction.html:
* LayoutTests/http/tests/resourceLoadStatistics/exemptDomains/app-bound-domains-exempt-from-cookie-blocking-between-each-other.html:
* LayoutTests/http/tests/resourceLoadStatistics/exemptDomains/managed-domains-third-party-resources-exempt-from-cookie-blocking.html:
* LayoutTests/http/tests/resourceLoadStatistics/exemptDomains/third-party-cookie-blocking.html:
* LayoutTests/http/tests/resourceLoadStatistics/grandfathering.html:
* LayoutTests/http/tests/resourceLoadStatistics/no-third-party-cookie-blocking-when-itp-is-off.html:
* LayoutTests/http/tests/resourceLoadStatistics/non-prevalent-resources-can-access-cookies-in-a-third-party-context.html:
* LayoutTests/http/tests/resourceLoadStatistics/ping-to-prevalent-resource.html:
* LayoutTests/http/tests/resourceLoadStatistics/resources/secure-worker-importing-localhost-script.js: Added.
(onmessage):
* LayoutTests/http/tests/resourceLoadStatistics/resources/set-cookie.py:
* LayoutTests/http/tests/resourceLoadStatistics/third-party-cookie-blocking-on-sites-without-user-interaction.html:
* LayoutTests/platform/glib/TestExpectations:
* LayoutTests/platform/gtk/TestExpectations:
* LayoutTests/platform/ios/TestExpectations:
* LayoutTests/platform/mac-wk2/TestExpectations:
* LayoutTests/platform/wincairo/TestExpectations:
* LayoutTests/platform/wk2/TestExpectations:

Canonical link: https://commits.webkit.org/280300@main
